src/main/java/org/candlepin/insights/validator/ip/IpAddressValidator.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.insights.validator.ip;
+
+import java.net.InetAddress;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+/**
+ * A ConstraintValidator that ensures that an IP is a valid IPV4 or IPV6 IP.
+ */
+public class IpAddressValidator implements ConstraintValidator<IpAddress, String> {
+
+    @Override
+    public void initialize(IpAddress constraintAnnotation) {
+    }
+
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        // A null or empty value is considered invalid.
+        if (value == null || value.isEmpty()) {
+            return false;
+        }
+
+        // InetAddress.getByName will try to resolve a hostname, but not if we pass it an actual
+        // IP. Do our best to determine whether we've receive what looks to be a hostname vs an IP.
+        // If we have a hostname, we consider the IP invalid.
+        if (isHostName(value)) {
+            return false;
+        }
+
+        try {
+            InetAddress.getByName(value);
```

Comment:
Let's just include `guava` and use [InetAddresses](https://google.github.io/guava/releases/19.0/api/docs/com/google/common/net/InetAddresses.html).  `gauva` is a useful library to have available anyway and from reading about `InetAddress` it looks like a malformed string can actually end up causing DNS look-ups which we don't want.

Reply:
I'll take a look. Thanks for the feedback.

Reply:
@awood MUCH MUCH better! Thanks for a great suggestion! :+1: 

Changes pushed! LGTM!